allenact_plugins/manipulathor_plugin/manipulathor_task_samplers.py

```diff
@@ -0,0 +1,496 @@
+        self.env_args = env_args
+        self.scenes = scenes
+        self.grid_size = 0.25
+        self.env: Optional[ManipulaTHOREnvironment] = None
+        self.sensors = sensors
+        self.max_steps = max_steps
+        self._action_space = action_space
+        self.objects = objects
+
+        self.scene_counter: Optional[int] = None
+        self.scene_order: Optional[List[str]] = None
+        self.scene_id: Optional[int] = None
+        self.scene_period: Optional[
+            Union[str, int]
+        ] = scene_period  # default makes a random choice
+        self.max_tasks: Optional[int] = None
+        self.reset_tasks = max_tasks
+
+        self._last_sampled_task: Optional[Task] = None
+
+        self.seed: Optional[int] = None
+        self.set_seed(seed)
+
+        if deterministic_cudnn:
+            set_deterministic_cudnn()
+
+        self.reset()
+        self.visualizers = visualizers
+        self.sampler_mode = kwargs["sampler_mode"]
+        self.cap_training = kwargs["cap_training"]
+
+    def _create_environment(self, **kwargs) -> ManipulaTHOREnvironment:
+        env = ManipulaTHOREnvironment(
+            make_agents_visible=False,
+            object_open_speed=0.05,
+            env_args=self.env_args,
+        )
+
+        return env
+
+    @property
+    def last_sampled_task(self) -> Optional[Task]:
+        return self._last_sampled_task
+
+    def close(self) -> None:
+        if self.env is not None:
+            self.env.stop()
+
+    @property
+    def all_observation_spaces_equal(self) -> bool:
+        """Check if observation spaces equal.
+
+        # Returns
+
+        True if all Tasks that can be sampled by this sampler have the
+            same observation space. Otherwise False.
+        """
+        return True
+
+    def reset(self):
+        self.scene_counter = 0
+        self.scene_order = list(range(len(self.scenes)))
+        random.shuffle(self.scene_order)
+        self.scene_id = 0
+        self.sampler_index = 0
+
+        self.max_tasks = self.reset_tasks
+
+    def set_seed(self, seed: int):
+        self.seed = seed
+        if seed is not None:
+            set_seed(seed)
+
+
+class SimpleArmPointNavGeneralSampler(AbstractMidLevelArmTaskSampler):
+
+    _TASK_TYPE = AbstractPickUpDropOffTask
+
+    def __init__(self, **kwargs) -> None:
+
+        super(SimpleArmPointNavGeneralSampler, self).__init__(**kwargs)
+        self.all_possible_points = []
+        for scene in self.scenes:
+            for object in self.objects:
+                valid_position_adr = "datasets/apnd-dataset/valid_object_positions/valid_{}_positions_in_{}.json".format(
+                    object, scene
+                )
+                try:
+                    with open(valid_position_adr) as f:
+                        data_points = json.load(f)
+                except Exception:
+                    print("Failed to load", valid_position_adr)
+                    continue
```

Comment:
Shouldn't this exception be fatal?

Reply:
Some other parts of the codebase have this problem too but we also need to find a better way to handle dataset paths that might not exist. Maybe add a `TODO` saying that this should be cleaned up.

Reply:
Well, this is in case an object does not exist in a room. It's not fatal. Just the dataset won't include that object in that specific room. I can log errors if you think that makes more sense.

Reply:
I think logging errors would make sense to me, maybe just replacing this with `get_logger().error(f"Failed to load {valid_position_adr}")` would be good.